pkg/common/aws/ec2.go

```diff
@@ -79,3 +85,49 @@ func (CcsAwsSession *ccsAwsSession) ReleaseElasticIPs(dryrun bool, sendSummary b
 
 	return nil
 }
+
+// TerminateEC2Instances finds EC2 instances with tag Name "osde2e*",
+// older than given duration, then terminates these EC2 instances.
+// Ignores EC2 instances with tag Name "osde2e-proxy*".
+func (CcsAwsSession *ccsAwsSession) TerminateEC2Instances(olderthan time.Duration, dryrun bool, sendSummary bool,
+	terminatedCounter *int, failedCounter *int, errorBuilder *strings.Builder,
+) error {
+	err := CcsAwsSession.GetAWSSessions()
+	if err != nil {
+		return err
+	}
+	result, err := CcsAwsSession.ec2.DescribeInstances(&ec2.DescribeInstancesInput{})
+	if err != nil {
+		return err
+	}
+	for _, reservation := range result.Reservations {
+		for _, instance := range reservation.Instances {
+			if time.Since(*instance.LaunchTime) > olderthan {
+				for _, tag := range instance.Tags {
+					if *tag.Key == "Name" && !strings.Contains(*tag.Value, osde2eproxysubstr) && strings.Contains(*tag.Value, osde2esubstr) {
+						fmt.Printf("Instance will be terminated: %s\n", instance)
+						if !dryrun {
+							input := &ec2.TerminateInstancesInput{
+								InstanceIds: []*string{instance.InstanceId},
+							}
+							_, err := CcsAwsSession.ec2.TerminateInstances(input)
+							if err != nil {
+								*failedCounter++
+								errorMsg := fmt.Sprintf("Instance %s failed to terminate: %s\n", *instance.InstanceId, err.Error())
+								fmt.Println(errorMsg)
+								if sendSummary && errorBuilder.Len() < config.SlackMessageLength {
```

Comment:
is there a case where this is true?

Reply:
As long as sendSummary is enabled and there are few to no errors.  The total slack notification length has to be under limits.